src/api/api.ts

```diff
@@ -289,50 +286,54 @@ export const listLinkedIssues = async (
   });
 };
 
-export const createIssue = async (
+export const getFields = async (
   client: IDeskproClient,
-  data: IssueFormData,
-  meta: Record<TransfornedFieldMeta["key"], TransfornedFieldMeta>,
-) => {
-  const customFields = {
-    ...Object.keys(data.customFields ?? []).reduce((fields, key) => {
-      const value = formatCustomFieldValue(meta[key], data.customFields?.[key]);
-
-      if (value === undefined) {
-        return fields;
-      }
+  settings?: Settings,
+): Promise<FieldMeta[]> => {
+  const isAdmin = Boolean(settings?.domain);
+  const baseUrl = isAdmin
+    ? `https://${settings?.domain}.atlassian.net/rest/api/3/field`
+    : `${API_BASE_URL}/field`;
 
-      return {
-        ...fields,
-        [key]: value,
-      };
-    }, {}),
-  };
+  const res = await request<FieldMeta[]>(client, "GET", baseUrl, undefined, settings);
 
-  // eslint-disable-next-line no-prototype-builtins
-  if (data.hasOwnProperty("customFields")) {
-    delete data.customFields;
-  }
+  return [
+    ...res,
+    ...(isAdmin ? [] : [
+      {
+        id: "linkedCount",
+        name: "Linked Issues",
+        key: "linkedCount",
+        schema: {
+          type: "string",
+          custom: "custom",
+        },
+      },
+      {
+        id: "key",
+        name: "Key",
+        key: "key",
+        schema: {
+          type: "string",
+          custom: "custom",
+        },
+      }
+    ]),
+  ] as FieldMeta[];
```

Comment:
Instead of `as` can we use [satisfies](https://tusharf5.com/posts/typescript-satisfies-operator/)

Reply:
Here, we need to explicitly cast the type. Because we manually add two elements to the array here and then it no longer matches the type. This is a specificity of the current implementation of custom fields.

Reply:
Can you do 
```
const data: FieldMeta[] = isAdmin ? [] : [...];
```
instead.

This will avoid the `as` type cast and ensure the code is safer.

Reply:
fixed